### PR TITLE
CI: add Kodiak config

### PR DESCRIPTION
**Pull request by nglgzz**
Added a base configuration for Kodiak to auto-merge PRs that have been approved and that passed CI checks.

### CODE REVIEW

**Comment by nikitavoloboev** [comment]
Dont we want to approve PRs before merging them?

**Comment by nikitavoloboev** [comment]
Will it squash and merge the PR?

**Comment by nikitavoloboev** [comment]
I think it would be good to have a history like in https://github.com/gatsbyjs/gatsby/commits/master

Where all commits are squashed merges with commit name as `PR name  (#<number>)` and description as PR description + list of commits perhaps.

What do you think?

**Comment by nglgzz** [comment]
It will rebase and merge. Just realized that this option is disabled in this repo though. Will update to squash instead. 

Yeah definitely, that would be cool, will look into it.

File: .kodiak.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+# use this for status checks that run indefinitely, like deploy jobs or the
+# WIP GitHub App
+dont_wait_on_status_checks = [] # default: [], options: list of check names (e.g. ["ci/circleci: lint_api"])
+
+
+[merge.message]
+# by default, github uses the first commit title for the PR of a merge.
+# "pull_request_title" uses the PR title.
+title = "github_default" # default: "github_default", options: "github_default", "pull_request_title"
```
Comment (johnletey) [inline .kodiak.toml]: @nglgzz If you want to do what @nikitavoloboev said, change this ling:

```suggestion
title = "pull_request_title" # default: "github_default", options: "github_default", "pull_request_title"
```

File: .kodiak.toml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# if this title regex matches, Kodiak will not merge the PR. this is useful
+# to prevent merging work in progress PRs
+blacklist_title_regex = "" # default: "^WIP.*", options: "" (disables regex), a regex string (e.g. ".*DONT\s*MERGE.*")
+
+# if these labels are set Kodiak will not merge the PR
+blacklist_labels = ["wip"] # default: [], options: list of label names (e.g. ["wip"])
+
+# choose a merge method. If the configured merge method is disabled for a
+# repository, Kodiak will report an error in a status message.
+method = "rebase" # default: "merge", options: "merge", "squash", "rebase"
```
Comment (johnletey) [inline .kodiak.toml]:
```suggestion
method = "squash" # default: "merge", options: "merge", "squash", "rebase"
```

**Comment by nglgzz** [comment]
Updated

File: .kodiak.toml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# once a PR is merged into master, delete the branch
+delete_branch_on_merge = true # default: false
+
+# if you request review from a user, don't merge until that user provides a
+# review, even if the PR is passing all checks
+block_on_reviews_requested = true # default: false
+
+# if there are running status checks on a PR when it's up for merge, don't
+# wait for those to finish before updating the branch
+optimistic_updates = true # default: true
```
Comment (johnletey) [inline .kodiak.toml]: We should definitely wait for checks to pass before merging @nglgzz 

```suggestion
optimistic_updates = false # default: true
```

File: .kodiak.toml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# .kodiak.toml
+version = 1
+
+[merge]
+
```
Comment (johnletey) [inline .kodiak.toml]: I don't like unneeded whitespace 😄

```suggestion
```

File: .kodiak.toml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+block_on_reviews_requested = true # default: false
+
+# if there are running status checks on a PR when it's up for merge, don't
+# wait for those to finish before updating the branch
+optimistic_updates = false # default: true
+
+# use this for status checks that run indefinitely, like deploy jobs or the
+# WIP GitHub App
+dont_wait_on_status_checks = [] # default: [], options: list of check names (e.g. ["ci/circleci: lint_api"])
+
```
Comment (johnletey) [inline .kodiak.toml]:
```suggestion
```

**Comment by nglgzz** [comment]
Fixed, need to approve again :smile:

**Pull request merged**